helix-term/src/handlers/completion.rs

```diff
@@ -291,6 +300,213 @@ fn request_completion(
+            #[cfg(unix)]
+            let mut path = {
+                static HOME_DIR: Lazy<Option<std::ffi::OsString>> =
+                    Lazy::new(|| std::env::var_os("HOME"));
+
+                let home_resolved_path = if let Some(home) = &*HOME_DIR {
+                    let first_separator_after_home = if matched_path.starts_with("~/") {
+                        Some(1)
+                    } else if matched_path.starts_with("$HOME") {
+                        Some(5)
+                    } else if matched_path.starts_with("${HOME}") {
+                        Some(7)
+                    } else {
+                        None
+                    };
+                    if let Some(start_char) = first_separator_after_home {
+                        let mut path = home.to_owned();
+                        path.push(&matched_path[start_char..]);
+                        path
+                    } else {
+                        matched_path.into()
+                    }
+                } else {
+                    matched_path.into()
+                };
+                PathBuf::from(home_resolved_path)
+            };
+            #[cfg(not(unix))]
+            let mut path = PathBuf::from(matched_path);
+
+            if path.is_relative() {
+                if let Some(doc_path) = doc.path().and_then(|dp| dp.parent()) {
+                    path = doc_path.join(path);
+                } else if let Ok(work_dir) = std::env::current_dir() {
+                    path = work_dir.join(path);
+                }
+            }
+            let ends_with_slash = match matched_path.chars().last() {
+                Some(std::path::MAIN_SEPARATOR) => true,
+                None => return None,
+                _ => false,
+            };
+            // check if there are chars after the last slash, and if these chars represent a directory
+            match std::fs::metadata(path.clone()).ok() {
+                Some(m) if m.is_dir() && ends_with_slash => {
+                    Some((PathBuf::from(path.as_path()), None))
+                }
+                _ if !ends_with_slash => path.parent().map(|parent_path| {
+                    (
+                        PathBuf::from(parent_path),
+                        path.file_name().and_then(|f| f.to_str().map(String::from)),
+                    )
+                }),
+                _ => None,
+            }
+        })?;
+
+    // The async file accessor functions of tokio were considered, but they were a bit slower
+    // and less ergonomic than just using the std functions in a separate "thread"
+    let future = tokio::task::spawn_blocking(move || {
+        let Some(read_dir) = std::fs::read_dir(&dir_path).ok() else {
+            return Vec::new();
+        };
+
+        read_dir
+            .filter_map(Result::ok)
+            .filter_map(|dir_entry| dir_entry.metadata().ok().map(|md| (dir_entry.path(), md)))
+            .map(|(path, md)| {
+                let file_name = path
```

Comment:
this is a bit ugly, what `dir_entry.path()` does is just `dir_path.join(dir_entry.file_name())` so going trough all this to extractd the file name again doesn't seem ideal. Instead just extract the file name and use that here (and just do `dir_path.join(file_name)` below)

Reply:
Should be handled in the recent push.